### PR TITLE
Introduce proto_gen

### PR DESCRIPTION
**Pull request by mssun**
## Description

Introduce `proto_gen` to help us generate proto files.

I remove the customized base64 serializer/deserializer because it will makes the issue more complicated. I understand base64 encoded u8 slices looks better, but we can continue to improve this later.

## Type of change (select applied and DELETE the others)

- [x] Rewrite proto generation logic

## How Has This Been Tested?

CI: https://ci.mesalock-linux.org/mssun/incubator-mesatee/186

## Checklist (check ALL before submitting PR, even not applicable)

- [x] Fork the repo and create your branch from `master`.
- [x] If you've added code that should be tested, add tests.
- [x] If you've changed APIs, update the documentation.
- [x] Ensure the tests pass (see CI results).
- [x] Make sure your code lints/format.

### CODE REVIEW

**Comment by mssun** [comment]
@litongxin1991 Can you take some time to review this first? I'd like to work on polishing KMS proto once this is merged. Thanks.

File: mesatee_services/kms/proto/build.rs [hunk 1]
```diff
@@ -15,33 +15,34 @@
...
 
-#[cfg(not(feature = "mesalock_sgx"))]
 fn main() {
+    let out_dir = env::var("OUT_DIR").expect("$OUT_DIR not set. Please build with cargo");
     println!("cargo:rerun-if-changed=src/kms.proto");
-    let src = PathBuf::from("src");
-    let output = src.join("prost_generated");
-    if !output.exists() {
-        std::fs::create_dir(&output).expect("failed to create prost_generated dir");
+    println!("cargo:rerun-if-changed=build.rs");
```
Comment (litongxin1991) [inline mesatee_services/kms/proto/build.rs]: Should we rerun the proto_gen if kms.proto changed?

Comment (mssun) [reply mesatee_services/kms/proto/build.rs]: In line 24: https://github.com/apache/incubator-teaclave/pull/180/files/ee9d73f9c8c54a37a38a9b61d54b124756bb1e61#diff-60cfea916835c29fdfe51e75a2a55a0eR24

File: mesatee_services/kms/proto/build.rs [hunk 2]
```diff
@@ -15,33 +15,34 @@
...
+    let c = Command::new("cargo")
```
Comment (litongxin1991) [inline mesatee_services/kms/proto/build.rs]: Will the cargo use vendored  libraries?

File: mesatee_services/kms/proto/src/lib.rs [hunk 3]
```diff
@@ -19,34 +19,14 @@
...
-    pub fn deserialize<'de, D>(deserializer: D) -> Result<Vec<u8>, D::Error>
-    where
-        D: Deserializer<'de>,
-    {
-        let s = <&str>::deserialize(deserializer)?;
-        base64::decode(s).map_err(de::Error::custom)
-    }
+pub mod proto {
+    #![allow(clippy::all)]
+    include!(concat!(env!("OUT_DIR"), "/kms_proto.rs"));
```
Comment (litongxin1991) [inline mesatee_services/kms/proto/src/lib.rs]: If I want to serialize and deserialize the ```vec<u8>``` with base64, should I add the serializer and  deserializer here or in ```prost_common```

Comment (mssun) [reply mesatee_services/kms/proto/src/lib.rs]: It's better to put in the generator.

BTW, I suggest to use hex to encode/decode u8. Firstly, no need to include third-party crates (`base64`) and we already have a implementation in `teaclave_config`. Secondly, it reads better than base64.

**Comment by litongxin1991** [comment]
I think we can remove the ```prost_generated``` now and also delete the some cmake commands for ignoring the ```prost_generated``` when ```make format``` is called, including
https://github.com/apache/incubator-teaclave/blob/ee9d73f9c8c54a37a38a9b61d54b124756bb1e61/cmake/UtilTargets.cmake#L10
https://github.com/apache/incubator-teaclave/blob/ee9d73f9c8c54a37a38a9b61d54b124756bb1e61/cmake/UtilTargets.cmake#L21

File: mesatee_services/kms/proto/src/lib.rs (hunk 3 above)
Comment (litongxin1991) [reply mesatee_services/kms/proto/src/lib.rs]: Thx. I agree. hex will be better.

**Comment by mssun** [comment]
Fixed. Latest CI: https://ci.mesalock-linux.org/mssun/incubator-mesatee/188. Please review.

File: mesatee_services/kms/proto/build.rs (hunk 1 above)
Comment (litongxin1991) [reply mesatee_services/kms/proto/build.rs]: Thanks

File: mesatee_services/kms/proto/build.rs (hunk 2 above)
Comment (mssun) [reply mesatee_services/kms/proto/build.rs]: No, an issue has been submitted: https://github.com/apache/incubator-teaclave/issues/181

**Comment by litongxin1991** [comment]
LGTM

**Pull request merged**